src/transforms/mapperTransforms.ts

```diff
@@ -462,20 +462,31 @@ function transformByteArrayMapper(pipelineValue: PipelineValue) {
 function transformChoiceMapper(pipelineValue: PipelineValue) {
   const { schema, options } = pipelineValue;
 
-  if (!isSchemaType([SchemaType.SealedChoice], schema)) {
+  if (!isSchemaType([SchemaType.SealedChoice, SchemaType.Choice], schema)) {
     return pipelineValue;
   }
 
   const choiceSchema = schema as ChoiceSchema;
-
-  const mapper = buildMapper(
-    schema,
-    {
+  let type;
+
+  if (isSchemaType([SchemaType.Choice], schema)) {
+    if (choiceSchema.choiceType && choiceSchema.choiceType.type) {
```

Comment:
According to the type definition of `ChoiceSchema` and `SealedChoiceSchema` it looks like `choiceType` and `choiceType.type` should never be undefined. Did you encounter any undefined during your testing?

Reply:
If not, we could probably skip this validation and simplify to something like this

```typescript
  if (isSchemaType([SchemaType.Choice], schema)) {
    type = {
      name: getMapperTypeFromSchema(choiceSchema.choiceType.type)
    };
  } else {
    type = {
      name: MapperType.Enum,
      allowedValues: choiceSchema.choices.map(choice => choice.value)
    };
  }

  const mapper = buildMapper(schema, type, options);
```

Reply:
Yes. In one of the unit tests. In choiceSchema, choiceType is defined directly as String. It will not have any property called type.

Reply:
> If not, we could probably skip this validation and simplify to something like this
> 
> ```ts
>   if (isSchemaType([SchemaType.Choice], schema)) {
>     type = {
>       name: getMapperTypeFromSchema(choiceSchema.choiceType.type)
>     };
>   } else {
>     type = {
>       name: MapperType.Enum,
>       allowedValues: choiceSchema.choices.map(choice => choice.value)
>     };
>   }
> 
>   const mapper = buildMapper(schema, type, options);
> ```

:-) This was my original code. Unfortunately, because of the test failure, I added the if check and modified the code

Reply:
Talked offline with @sarangan12 , we should file a bug in M4 to either throw when ChoiceSchema is constructed without specifying the choiceType or make choiceType an optional property of ChoiceSchema